ctest-next/src/generator.rs

```diff
@@ -65,20 +105,371 @@ impl TestGenerator {
+        self
+    }
+
+    /// Set a `-D` flag for the C compiler being called.
+    ///
+    /// This can be used to define various variables to configure how header
+    /// files are included or what APIs are exposed from header files.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use ctest_next::TestGenerator;
+    ///
+    /// let mut cfg = TestGenerator::new();
+    /// cfg.define("_GNU_SOURCE", None)
+    ///    .define("_WIN32_WINNT", Some("0x8000"));
+    /// ```
+    pub fn define(&mut self, k: &str, v: Option<&str>) -> &mut Self {
+        self.defines
+            .push((k.to_string(), v.map(std::string::ToString::to_string)));
+        self
+    }
+
+    /// Configures how Rust `const`s names are translated to C.
+    pub fn rename_constant(&mut self, f: impl Fn(&Const) -> Option<String> + 'static) -> &mut Self {
+        self.mapped_names.push(Box::new(move |item| {
+            if let MapInput::Const(c) = item {
+                f(c)
+            } else {
+                None
+            }
+        }));
+        self
+    }
+
+    /// Configures how a Rust struct field is translated to a C struct field.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use ctest_next::TestGenerator;
+    ///
+    /// let mut cfg = TestGenerator::new();
+    /// cfg.rename_field(|_s, field| {
+    ///     Some(field.ident().replace("foo", "bar"))
+    /// });
+    /// ```
+    pub fn rename_field(
+        &mut self,
+        f: impl Fn(&Struct, &Field) -> Option<String> + 'static,
+    ) -> &mut Self {
+        self.mapped_names.push(Box::new(move |item| {
+            if let MapInput::Field(s, c) = item {
+                f(s, c)
+            } else {
+                None
+            }
+        }));
+        self
+    }
+
+    /// Configures the name of a function in the generated C code.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use ctest_next::TestGenerator;
+    ///
+    /// let mut cfg = TestGenerator::new();
+    /// cfg.rename_fn(|f| Some(format!("{}_c", f.ident())));
+    /// ```
+    pub fn rename_fn(&mut self, f: impl Fn(&crate::Fn) -> Option<String> + 'static) -> &mut Self {
+        self.mapped_names.push(Box::new(move |item| {
+            if let MapInput::Fn(func) = item {
+                f(func)
+            } else {
+                None
+            }
+        }));
+        self
+    }
+
+    /// Configures how a Rust type is translated to a C type.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use ctest_next::{TestGenerator, TyKind};
+    ///
+    /// let mut cfg = TestGenerator::new();
+    /// cfg.rename_type(|ty, kind| {
+    ///     if kind == TyKind::Struct {
+    ///         Some(format!("{}_t", ty))
+    ///     } else {
+    ///         None
+    ///     }
+    /// });
+    /// ```
+    pub fn rename_type(
+        &mut self,
+        f: impl Fn(&str, TyKind) -> Option<String> + 'static,
+    ) -> &mut Self {
```

Comment:
Similarly, could this be split to `rename_struct` and `rename_union`? So we shouldn't need a `TyKind`.